bento_lib/auth/wrappers.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+from flask import current_app
+
+from .middleware import AuthxFlaskMiddleware
+
+
+def authn_token_required_flask_wrapper(f):
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        # expecting a globally accessible instance of AuthxFlaskMiddleware
+        if current_app.authx is not None and \
```

Comment:
crashes if current_app.authx is `None`, an alternative is `current_app.get("authx")` (which returns None if it's missing)

Reply:
I don't see anything _forbidding_ pushing new properties to current_app, but common practice is to put things like this inside existing fields, like `current_app.config`.

Reply:
i probably wouldnt put an object in config, since config generally is restricted to literals like strings/ints...

there is however a specific way of adding pre-request hooks to flask to inject stuff into `g` (see `https://flask.palletsprojects.com/en/2.2.x/api/#flask.Flask.before_request`) which could call a verify_token function 

this could be structured as a constructor call to the middleware passing the application in, which then adds the before_request hook or something.

```py
app = Flask(...)

AuthxFlaskMiddleware(app, ...)  # adds the hook etc.

```

this is a similar pattern to how `Flask-CORS` does stuff: https://flask-cors.readthedocs.io/en/latest/

Reply:
The Flask `g` will certainly come in handy
Thus far, I aimed to distinguish 'app-wide' from 'individual-request-wide' by using `current_app` along with `g`